plugins/thelio-io/fu-thelio-io-device.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-io-channel.h"
+#include "fu-thelio-io-device.h"
+
+struct _FuThelioIoDevice {
+	FuUsbDevice			 parent_instance;
+};
+
+G_DEFINE_TYPE (FuThelioIoDevice, fu_thelio_io_device, FU_TYPE_USB_DEVICE)
+
+static gboolean
+fu_thelio_io_device_probe (FuDevice *device, GError **error)
+{
+	g_autofree gchar *fn = NULL;
+	g_autofree gchar *buf = NULL;
+	g_autoptr(GUdevDevice) udev_device = NULL;
+
+	fu_device_add_instance_id (device, "USB\\VID_03EB&PID_2FF4");
+
+	/* convert GUsbDevice to GUdevDevice */
+	udev_device = fu_usb_device_find_udev_device (FU_USB_DEVICE (device), error);
+	if (udev_device == NULL)
+		return FALSE;
+
+	fn = g_build_filename (g_udev_device_get_sysfs_path (udev_device), "revision", NULL);
```

Comment:
although unlikely, you probably should check that the path you get from `g_udev_device_get_sysfs_path` is non-NULL.  (Sample code from another plugin here: https://github.com/fwupd/fwupd/blob/master/plugins/thunderbolt/fu-plugin-thunderbolt.c#L168)